### DISCUSSION TITLE
is it possible to disable HAL/Arduino handling of a specific timer interrupt?

### DISCUSSION THREAD

**Discussion by PerennialNovice** [stm32duino/Arduino_Core_STM32#2321] [Q&A] [answered]
I saw that in HardwareTimer.cpp timer-ISR are surrounded by preprocessor conditions like `#if defined(TIM2_BASE)`
I tried using `#undef TIM2_BASE` in the arduino setup() functions but that did not work as hoped

Is there a possibility to disable certain timers (or part of other peripheral) instead of completely disabling like with HAL_TIM_MODULE_ONLY or HAL_TIM_MODULE_DISABLED?

**Comment by fpistm** [comment, marked as answer]
This is the CMSIS device definition (header file). When you undef in setup it is local to the cpp file (.ino). Even if you disable it globally, I do not advise as it is probably used by LL/HAL source files and CMSIS one.

Finally to answer your title question, no it is not possible. The only way is to use `HAL_TIM_MODULE_ONLY ` to globally disable the usage of the HAL_TIM_MODULE by the Arduino API level.

**Comment by PerennialNovice** [comment]
thanks!
